File: src/main/resources/static/app/service/ProdutoService.js
'user strict';

zombieApp.service('ProdutoService', ['ProdutoRepository',
    function (ProdutoRepository) {

        return {
            listarProdutos: function () {
                return ProdutoRepository.listarProdutos();
            },
            bid: function (produto) {
                return ProdutoRepository.bid(produto);
            }

        }
    }]);

File: src/main/java/com/redeempresarial/leilao/services/ProdutoService.java
package com.redeempresarial.leilao.services;

import com.redeempresarial.leilao.domain.Produto;

import java.util.List;

/**
 * Created by brunocassio on 25/04/17.
 */
public interface ProdutoService {

    List<Produto> listarProdutos();

    List<Produto> bid(Produto produto);
}
